src/schematic/schematic_rect.ts

```diff
@@ -0,0 +1,45 @@
+import { z } from "zod"
+import { point, type Point } from "../common/point"
+import { distance, rotation } from "../units"
+import { getZodPrefixedIdWithDefault } from "src/common"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+/** Draws a styled rectangle on the schematic */
+export interface SchematicRect {
+  type: "schematic_rect"
+  schematic_rect_id: string
+  schematic_component_id: string
+  center: Point
+  width: number
+  height: number
+  rotation: number
+  stroke_width: number
+  color: string
+  is_filled: boolean
+  fill_color?: string
+  is_dashed: boolean
+  subcircuit_id?: string
+}
+
+export const schematic_rect = z
+  .object({
+    type: z.literal("schematic_rect"),
+    schematic_rect_id: getZodPrefixedIdWithDefault("schematic_rect"),
+    schematic_component_id: z.string(),
+    center: point,
+    width: distance,
+    height: distance,
+    rotation: rotation.default(0),
+    stroke_width: distance.default(0.02),
```

Comment:
it's beneficial to have stroke_width allowed to be undefined to make it scale-invariant

```suggestion
    stroke_width: distance.nullable().optional(),
```